code/script.js

```diff
@@ -1,51 +1,285 @@
+    inputWrapper.innerHTML = ''
+    setTimeout (() => doYouNeedHelp('option2'), 1000)
+  })
+}
+
+//Question 3: Bot's answer depends on if user chooses Yes ( if (selection === 'option1') ) or No ( else if (selection === 'option2') ) in Question 2
+const doYouNeedHelp = (selection) => {
+  if (selection === 'option1') {
+    showMessage(`I'm happy to help! How are you feeling today?`, 'bot')
+    setTimeout (() => moodChoice (), 1000)
+    
+   const moodChoice = () =>  {
+    inputWrapper.innerHTML= 
+    `<button id="optHappy">😄</button>
+    <button id="optSad">😥</button>
+    <button id="optAngry">😡</button>
+    `
+   
+    document.getElementById("optHappy").addEventListener('click', () => {
+      showMessage(`I'm feeling happy today`, 'user')
+      inputWrapper.innerHTML = ''
+      setTimeout (() => chooseByMood('optHappy'), 1000)
+    })
+
+    document.getElementById("optSad").addEventListener('click', () => {
+      showMessage(`I'm feeling sad today`, 'user')
+      inputWrapper.innerHTML = ''
+      setTimeout (() => chooseByMood('optSad'), 1000)
+    })
+
+    document.getElementById("optAngry").addEventListener('click', () => {
+      showMessage(`I'm feeling angry today`, 'user')
+      inputWrapper.innerHTML = ''
+      setTimeout (() => chooseByMood('optAngry'), 1000)
+    })
+  }
+  } 
+  
+  else if (selection === 'option2') {
+    showMessage(`Well, now that you know it works, you're more than welcome back once you need my help!`, 'bot')
+    setTimeout (() => inputWrapper.innerHTML=`
+    <p>Thanks for using this bot!</p>
+    `, 1000)
+    }
+}
+
+//Question 4: Bot suggests movie. The bot's answer depends on which Mood-button has been chosen ('optHappy', 'optSad', 'optAngry'). 
+const chooseByMood = (selection) => {
+ if (selection === 'optHappy') {
+  showMessage(`Great to hear that you are happy! Then one of these movies might suit your mood`, 'bot')
+  setTimeout (() => happyFilmButtons (), 1000)
+  
+  const happyFilmButtons = () => {
+  inputWrapper.innerHTML=`
+      <button id="happyFeet">Happy feet</button>
+      <button id="singingInTheRain">Singing in the Rain</button>
+      <button id="greaseMovie">Grease</button>
+      `
+
+      document.getElementById("happyFeet").addEventListener('click', () => {
+        showMessage(`I'll watch Happy Feet tonight!`, 'user')
+        inputWrapper.innerHTML = ''
+        setTimeout (() => lastReplyHappy('happyFeet'), 1000)
+      })
+
+      document.getElementById("singingInTheRain").addEventListener('click', () => {
+        showMessage(`I'll watch Singing in the Rain tonight!`, 'user')
+        inputWrapper.innerHTML = ''
+        setTimeout (() => lastReplyHappy('singingInTheRain'), 1000)
+      })
+
+      document.getElementById("greaseMovie").addEventListener('click', () => {
+        showMessage(`I'll watch Grease tonight`, 'user')
+        inputWrapper.innerHTML = ''
+        setTimeout (() => lastReplyHappy('grease'), 1000)
+      }) 
+    }
```

Comment:
As with the earlier comment of row 75, we would also consider putting this in a separate function to ease reading.